app/models/notification.rb

```diff
@@ -55,56 +53,51 @@ def self.all_flavors
   #   naming::    Naming that triggered this email.
   #
   def calc_note(args)
-    if template = note_template
-      case flavor.to_sym
-      when :name
-        tracker  = user
-        observer = args[:user]
-        naming   = args[:naming]
-        unless observer
-          raise "Missing 'user' argument for #{flavor} notification."
-        end
-        unless naming
-          raise "Missing 'naming' argument for #{flavor} notification."
-        end
+    return unless (template = note_template)
 
-        template.
-          gsub(":observer", observer.login).
-          gsub(":observation", "#{MO.http_domain}/#{naming.observation_id}").
-          gsub(":mailing_address", tracker.mailing_address || "").
-          gsub(":location", naming.observation.place_name).
-          gsub(":name", naming.format_name)
+    case flavor.to_sym
+    when :name
```

Comment:
It's not at all clear to my why there are all these single case `case` statements.  I would convert all of these to `if`s.

Reply:
My guess:
There are 4 defined flavors, see line 35, so it was written as `case`. But only 1 flavor is available, so each `case` has just the single case.
I'll re-write as `if`.